src/main/java/com/siawsam/duke/Parser.java

```diff
@@ -0,0 +1,112 @@
+package com.siawsam.duke;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A parser for user inputs.
+ */
+public class Parser {
+    /**
+     * An array of valid Duke commands to edit tasks.
+     */
+    private static final List<String> COMMANDS = Arrays.asList("done", "delete", "find");
+    private final TaskList userTaskList;
+    private final Storage storage;
+    
+    /**
+     * Constructs a Parser when no saved task list exists.
+     *
+     * @param storage A Storage instance to use when the parser needs to save to disk.
+     */
+    public Parser(Storage storage) {
+        this.userTaskList = new TaskList();
```

Comment:
Is the 'this' keyword necessary here? The userTasklist variable is not shadowed by any constructor parameters here so I think it may be better to leave it out as per the coding standard guide. I noticed this is some other parts as well.

Reply:
You're absolutely right. I only learnt about this rule in the coding standard guide while reviewing other PRs too. Haven't gotten around to cleaning my code but I will moving forward!